### DISCUSSION TITLE
OpenProject - Open source project management software

### DISCUSSION THREAD

**Discussion by 391a** [community-scripts/ProxmoxVE#964] [Request script]
### Application Name

OpenProject

### Website

https://www.openproject.org/

### Description

OpenProject is the leading open source project management software.
https://github.com/opf/openproject

### Due Diligence

- [X] I have searched existing [scripts](https://community-scripts.github.io/Proxmox/scripts) and found no duplicates.
- [X] I have searched existing [discussions](https://github.com/community-scripts/ProxmoxVE/discussions?discussions_q=) and found no duplicate requests.
- [X] This is not a game-related request.

**Comment by skyparkjustin** [comment]
https://www.openproject.org/docs/installation-and-operations/installation/packaged/
Script setup should be straight forward.

**Comment by FinFox1** [comment]
Would also love to see this as an LXC script.

**Comment by michelroegl-brunner** [comment]
Ready for testing:

bash -c "$(wget -qLO - https://github.com/community-scripts/ProxmoxVED/raw/main/ct/openproject.sh)"

**Reply by FinFox1** [reply]
Many thanks for the quick response! Will try this one out and providing feedback later on.

**Reply by FinFox1** [reply]
So, first of all i wanna thank you very much for making this script available. The installation process works smooth and fast! Typically, when installing openProject, at the very beginning, you have the choice to select the edition. However, actually in the /etc/openproject/installer.dat there is the option "default". This one has to be switched to "bim" i guess in order to select between the editions. I have attached the two screenshots, so you can see yourself. After changing the edition form "default" to "bim" and ran "sudo openproject reconfigure" the selection of the editions became available.

![Screenshot 2025-04-03 043406](https://github.com/user-attachments/assets/c44d47fa-f7b1-4045-91c5-fb75fc9287e4)
![Screenshot 2025-04-03 043921](https://github.com/user-attachments/assets/6818cdd2-8794-4054-a577-e50816c48446)
<img width="1136" alt="Screenshot 2025-04-03 044552" src="https://github.com/user-attachments/assets/71c3593a-0de3-47c8-8d35-0282cd1c1d9d" />

**Reply by michelroegl-brunner** [reply]
Yes but bim is the construction version or? default installs the normal version.
I write a guide an like it on the frontend how to activate the bim version. 
If more people request the bim version we might change it.

**Reply by FinFox1** [reply]
> Yes but bim is the construction version or? default installs the normal version. I write a guide an like it on the frontend how to activate the bim version. If more people request the bim version we might change it.

That´s right, BIM stands for "Building Information Modeling" and enhances the OpenProject Platform with specific BIM features, meant for the construction industry. I just wanted to mentioned it and it is not very complicated to "reconfigure" or "reinstall" the OpenProject.

**Reply by Fischje** [reply]
How to change the default host_name configuration to put openprojekt after a reverse proxy? A reconfigrue procedure asks me for db-settings and things i dont know to configure.

I get:
_Invalid host_name configuration_

**Reply by MickLesk** [reply]
https://community.openproject.org/projects/openproject/forums

**Comment by MickLesk** [comment]
PR Created for Default Version - Other is an new LXC or an Enhancement-Feature-Request

**Comment by FinFox1** [comment]
I faced some problems, when installing and reconfiguring the LXC container with Debian 12. There are some invalid keyrings, which can be solved by using this command after installation:

**curl -fsSL https://packages.microsoft.com/keys/microsoft.asc | sudo gpg --dearmor -o /usr/share/keyrings/microsoft-prod.gpg**
